### PR TITLE
[connectors] chore(microsoft): Improve garbageCollector performances

### PR DESCRIPTION
**Pull request by tdraier**
## Description

- Use batch requests on microsoft (can take 20 requests per batch max)
- Fetch 300 nodes per iteration

## Deploy Plan

deploy `connectors`

### CODE REVIEW

File: connectors/src/connectors/microsoft/temporal/activities.ts [hunk 1]
```diff
@@ -903,69 +903,73 @@ export async function microsoftGarbageCollectionActivity({
...
-            rootNodeIds,
-            startGarbageCollectionTs,
-          }))
-        ) {
-          await deleteFile({
-            connectorId,
-            internalId: node.internalId,
-            dataSourceConfig,
-          });
+  const chunkedRequests = [];
```
Comment (philipperolet) [inline connectors/src/connectors/microsoft/temporal/activities.ts:912]: nit: you can use `_.chunk` there i think

File: connectors/src/connectors/microsoft/temporal/activities.ts [hunk 2]
```diff
@@ -903,69 +903,73 @@ export async function microsoftGarbageCollectionActivity({
...
+  while (requests.length) {
+    chunkedRequests.push(requests.splice(0, 20));
+  }
+
+  for (const chunk of chunkedRequests) {
+    const batchRes = await client.api("/$batch").post({ requests: chunk });
```
Comment (philipperolet) [inline connectors/src/connectors/microsoft/temporal/activities.ts:918]: just checking that when you tested in local the replies were correct

**Review by philipperolet — APPROVED** [review]
Looks good :pray: left 2 coms, pre-approved :)

File: connectors/src/connectors/microsoft/temporal/activities.ts (hunk 2 above)
Comment (tdraier) [reply connectors/src/connectors/microsoft/temporal/activities.ts:918]: yes i tried with existing files, and removed files (not sure if I went into 404 or deleted flag, but the file was garbage collected)

**Pull request merged**